src/librbd/CopyupRequest.cc

```diff
@@ -180,6 +180,26 @@ class UpdateObjectMap : public C_AsyncObjectThrottle<> {
     return false;
   }
 
+  bool CopyupRequest::is_nop()
+  {
+    if (!m_copyup_data.is_zero()) {
+      return false;
+    }
+
+    for (const AioObjectRequest *req : m_pending_requests) {
+      const AioObjectWrite *wreq = dynamic_cast<const AioObjectWrite *>(req);
```

Comment:
I'd avoid the dynamic cast and add a new ```AioObjectRequest::is_no_op``` virtual method (combines the write check and the ::all_zero check).